File: java/src/leetcode/easy/RemoveElement.java
package leetcode.easy;

/**
 * Question: https://leetcode.com/problems/remove-element/description/
 */
public class RemoveElement {
    public int removeElement(int[] nums, int val) {
        int i = 0, j = 0;
        while (j < nums.length) {
            if (nums[j] != val) {
                swap(nums, i, j);
                i++;
            }
            j++;
        }
        return i;
    }

    private void swap(int[] arr, int i, int j) {
        if (i != j) {
            arr[i] = arr[i] + arr[j];
            arr[j] = arr[i] - arr[j];
            arr[i] = arr[i] - arr[j];
        }
    }
}

File: ruby/easy/remove_element.rb
# @param {Integer[]} nums
# @param {Integer} val
# @return {Integer}
def remove_element(nums, val)
    i = 0
    for j in 0..nums.length-1
      if(nums[j] != val)
        nums[i] = nums[j]
        i += 1
      end
    end
    return i
end